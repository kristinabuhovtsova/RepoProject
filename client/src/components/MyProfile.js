import React from 'react';
import store from '../store'
import {NavLink} from 'react-router-dom'
import axios from 'axios';
import SignOutAction from '../actions/signOutAction';
import DeleteAction from '../actions/deleteAction';
import '../css/Button.css';

const Deletting = () => {
    const username = store.getState().in.username
    axios.delete ('/catalog/signup', {params: {'username': username}})
        .then (store.dispatch (SignOutAction()))
        .then (store.dispatch(DeleteAction(username)))
}

class Profile extends React.Component {
    constructor (props) {
    super (props)
    this.state = { countries: []}
   };
  
    componentDidMount() {
        const allstat=this.props.stat
        const name= this.props.username
        let viscountr=[]
        if (this.props.stat !== []) {
        allstat.forEach ((info) => {if (info.username === name){
            info.viscountries.map(country =>  viscountr.push(country))}})}
        this.setState ({countries: viscountr})}
    

    render () {
        if (this.props.in) {
            return (
                <div className = 'col-9'>
                    <h3>My profile</h3>
                    <div className='profile-button'>
                    <button className="btn btn-outline-primary signinbut" onClick={Deletting}>Delete profile </button>
                    </div>
                    <p>Name: <NavLink to='/changename'> {this.props.username}</NavLink> </p>
                    {this.state.countries && <p>My travels: </p>}
                    {this.state.countries && this.state.countries.map((country, ind) =>
                    <p key={ind}>{ind+1} 
                    <NavLink to={String('/countries/'+ country)}> {country}</NavLink>
                    </p>
                    )}
                </div>
            )}
        else {
            return (<h3>Firstly, you need to sign in</h3>)
        }
        }
}

export default Profile