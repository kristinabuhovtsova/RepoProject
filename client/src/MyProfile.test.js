import React from 'react';
import MyProfile from './components/MyProfile'
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

it("correct render of profile", () => {
    const statist=[{username: 'john13', usercountry: 'Canada', viscountries: ['Japan', 'Italy', 'Sweeden'] },
    {username: 'frank13', usercountry: 'Canada', viscountries: ['France', 'Albania'] }
]
    const tree = renderer
          .create(  <BrowserRouter><MyProfile in={1} stat={statist} username={'john13'}/></BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot(); 
});

