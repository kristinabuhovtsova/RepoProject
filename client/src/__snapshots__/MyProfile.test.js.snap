// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct render of profile 1`] = `
<div
  className="col-9"
>
  <h3>
    My profile
  </h3>
  <div
    className="profile-button"
  >
    <button
      className="btn btn-outline-primary signinbut"
      onClick={[Function]}
    >
      Delete profile 
    </button>
  </div>
  <p>
    Name: 
    <a
      aria-current={null}
      href="/changename"
      onClick={[Function]}
    >
       
      john13
    </a>
     
  </p>
  <p>
    My travels: 
  </p>
  <p>
    1
    <a
      aria-current={null}
      href="/countries/Japan"
      onClick={[Function]}
    >
       
      Japan
    </a>
  </p>
  <p>
    2
    <a
      aria-current={null}
      href="/countries/Italy"
      onClick={[Function]}
    >
       
      Italy
    </a>
  </p>
  <p>
    3
    <a
      aria-current={null}
      href="/countries/Sweeden"
      onClick={[Function]}
    >
       
      Sweeden
    </a>
  </p>
</div>
`;
